accepted/wasmtime-one-dot-oh.md

```diff
@@ -0,0 +1,558 @@
+WebAssembly's use in mission- and security-critical environments, and we believe
+that we need to hold ourselves to higher standards accordingly.
+
+## What does it mean for a feature to be stable?
+
+The tiers of support for Wasmtime are primarily concerned with the APIs that
+users interact with and the support expected there, but Wasmtime is also
+defined by the stability of its implementation. Features implemented in Wasmtime
+itself are held to a high standard of quality which should fit into the
+quality expectations of components that are tier 1. Features enabled-by-default
+and implemented in Wasmtime are expected to meet some baseline criteria:
+
+* The feature must be throughly tested in Wasmtime's CI on major platforms (at
+  this time x86\_64 and AArch64)
+
+* If the feature is for a WebAssembly upstream proposal, all spec tests must be
+  enabled and passing and the proposal must be in [stage 4 or
+  later](https://github.com/WebAssembly/meetings/blob/main/process/phases.md).
+
+* The feature must have no open questions, design concerns, or serious known
+  bugs.
+
+* The feature must have support in fuzzers and have been fuzzed for at least a
+  week. Additionally we should be confident that the fuzzing support for this
+  feature is exercising the necessary bits in Wasmtime thoroughly.
+
+* The feature is supported in the Rust API, the C API, and at least one other
+  language's embedding API.
+
+* A member of the Bytecode Alliance must be "on the hook" for maintenance of
+  this feature.
+
+Features can be implemented in-tree even if they do not meet these criteria but
+the features must be disabled-by-default at either runtime or compile
+time. If a feature's interim implementation does not have an undue compile-time
+or runtime footprint then it can be off-by-default at runtime but compiled in by
+default. If, however, an in-progress feature has a significant compile-time or
+runtime footprint it must be disabled at compile-time by default.
+
+Features implemented in-tree but not currently stabilized are also subject to
+removal if there is no active progress being made on the feature. For example if
+this RFC is approved **the `lightbeam` feature of Wasmtime would be removed**.
+
+> **Note**: This RFC proposes removing `lightbeam` because it isn't actively
+> maintained, and as a result hasn't been compiling successfully for quite
+> some time. The Wasmtime project is still interested in the addition of a
+> baseline compiler, but the addition of one would be proposed in an RFC and
+> undergo careful evaluation to ensure the design aligns with Wasmtime's
+> goals and requirements, and the compiler has a solid maintenance story.
+
+## Release Process and Cadence
+
+Wasmtime intends to follow in the footsteps of many other projects on the matter
+of release cadence with a frequent and predictable release cycle. **This RFC
+proposes releasing Wasmtime on Tuesday every 4 weeks**. The precise date of
```

Comment:
Monthly releases (e.g. releasing Wasmtime on the first Tuesday of every month) might be easier to reason about, seeing as even this document uses "once a month" and "every 4 weeks" interchangebly.

Reply:
My main goal is to make sure that the releases are as automate-able as possible to make the work of a release on us in theory "make sure the changelog is up-to-date and merge a PR". In that sense I don't think that the precise cadence really matters too too much. I'm expecting users will have some sort of regular process to update `wasmtime` and/or notifications through some bot like dependabot.

All that to say I think the precise number doesn't really matter too too much and I think that whatever is the easiest and most reliable to automate is probably the way to go. I chose a weekly-based cadence since that's what rust-lang/rust does (although 6 weeks instead of 4) and I know there's a fair bit of tooling for things like "when's the next release?" which I believe is easier with a machine-predictable cadence rather than a human-predictable cadence. I don't think it's really onerous one way or another though.